types/index.d.ts

```diff
@@ -38,6 +38,58 @@ declare function xastscript(
   name: string,
   attributes?: Attributes,
   ...children: Children[]
-): Element
+): xast.Element
+
+/**
+ * This unique symbol is declared to specify the key on which JSX children are passed, without conflicting
+ * with the Attributes type.
+ */
+declare const children: unique symbol
+
+/**
+ * This namespace allows to use `xastscript` as a JSX implementation.
+ *
+ * This namespace is only used to support the use as JSX. It’s **not** intended for direct usage.
+ */
+declare namespace xastscript.JSX {
+  /**
+   * This defines the return value of JSX syntax.
+   */
+  type Element = xast.Element | xast.Root
+
+  /**
+   * This disallows the use of functional components.
+   */
+  type IntrinsicAttributes = never
+
+  /**
+   * This defines the prop types for known elements.
+   *
+   * For `xastscript` this defines any string may be used in combination with `xast` `Attributes`.
+   *
+   * This **must** be an interface.
+   */
+  // eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
+  interface IntrinsicElements {
+    [tagName: string]:
+      | Attributes
+      | {
+          /**
+           * The prop that matches `ElementChildrenAttribute` key defines the type of JSX children, defines the children type.
+           */
+          [children]?: Children
+        }
+  }
+
+  /**
+   * The key of this interface defines as what prop children are passed.
+   */
+  interface ElementChildrenAttribute {
+    /**
+     * Only the key matters, not the value.
+     */
+    [children]?: any
```

Comment:
if only the key matters, would it be safe to use `unknown` here?

Reply:
I thought I did, but apparently I didn’t :thinking: 

I changed it to `never`, which I think fits even better than `unknown`.